### ISSUE TITLE
Final convergence: route Sampled/Gumbel MuZero, fix continuous MCTS, strengthen smoke tests

### ISSUE THREAD

**Issue by w2030298-art** [w2030298-art/muzero-rebuilt#1]
## Goal

Bring `master` to final delivery quality by fixing the gaps found in the GitHub quick review:

1. `sampled_muzero` / continuous action search must be routed from config and execute through MCTS.
2. `gumbel_muzero` must actually use `GumbelPolicy` for root sampling / selection.
3. Pendulum sampled and Gumbel integration tests must run real self-play episodes, not only component-build checks.
4. README / progress / issues docs must match the actual implementation and verification state.
5. Full lint / type / test / train / benchmark gates must pass.

### Module A: Config routing

- Modify `src/muzero/cli/train.py`.
- Add `_build_action_sampler(config, env_spec)`:
  - `discrete` -> `DiscreteActionSampler()`.
  - `continuous` -> `ContinuousActionSampler(action_low=np.asarray(env_spec.action.low, dtype=np.float32), action_high=np.asarray(env_spec.action.high, dtype=np.float32), sampling_std=0.3)`.
  - If low/high is missing, raise `ValueError("continuous action sampler requires action low/high bounds")`.
- Add `_build_search_policy(config)`:
  - `config.algorithm.name == "gumbel_muzero"` -> `GumbelPolicy(num_root_samples=config.search.num_sampled_actions)`.
  - otherwise -> `PUCTPolicy()`.
- Replace fixed `DiscreteActionSampler()` / `PUCTPolicy()` construction with these factories.
- Pass `action_space_type=env_spec.action.type` to `MCTS`.

### Module B: Continuous-action MCTS

- Modify `src/muzero/search/mcts.py`.
- Add `action_space_type: str = "discrete"` to `MCTS.__init__()` and store `self._action_space_type`.
- Add `_action_to_tensor(self, action: Action) -> torch.Tensor`:
  - discrete -> `torch.tensor([int(action)], device=self._device, dtype=torch.long)`.
  - continuous -> `np.asarray(action, dtype=np.float32).reshape(1, -1)` then `torch.from_numpy(arr).to(self._device)`.
  - if continuous dim mismatches `self._action_dim`, raise `ValueError`.
- In `_expand_leaf()`, replace `torch.tensor([int(action)], ...)` with `self._action_to_tensor(action)`.
- In `_recurrent_inference()`, if batcher exists but `self._action_space_type == "continuous"`, bypass batcher and call network directly.
- Preserve discrete batcher behavior.
- Ensure continuous `SearchResult.sampled_actions` is populated from root child actions.

### Module C: Gumbel real search integration

- Modify `src/muzero/search/mcts.py`.
- Import `GumbelPolicy`.
- In `_expand_root()`, when `self._search_policy` is `GumbelPolicy`:
  - call `sample_root_actions(policy_logits=policy_output[0], legal_mask=legal_mask_tensor, num_samples=self._config.num_sampled_actions)`.
  - create only those root children.
  - compute priors from `softmax(policy_output[0])` and renormalize selected priors.
- In `run()`, if policy is `GumbelPolicy`, select root action with `sequential_halving(tree, root_id)`; otherwise keep `select_root_action()`.

### Module D: Strengthen integration tests

- Modify `tests/integration/test_pendulum_sampled_smoke.py`:
  - add a test that loads `configs/pendulum_sampled_muzero.yaml`, lowers `num_simulations` to 2, `num_sampled_actions` to 4, `max_episode_steps` to 3, builds components, asserts `ContinuousActionSampler`, runs `SelfPlayWorker.run_episode(seed=0)`, and asserts all actions are `np.ndarray` within env action bounds.
- Modify `tests/integration/test_gumbel_smoke.py`:
  - add a test that loads `configs/tictactoe_gumbel_muzero.yaml`, lowers `num_simulations` to 2, `num_sampled_actions` to 4, `max_episode_steps` to 3, builds components, asserts `GumbelPolicy`, runs `SelfPlayWorker.run_episode(seed=0)`, and asserts all actions are integers in `0..8`.

### Module E: Sync docs

- Update `README.md` so it no longer says `Work in progress`.
- Add quick verification commands.
- Update `docs/progress.md` to show final verification / accurate module status.
- Update `docs/issues.md` with the GitHub quick review fix summary.
- Do not claim commands passed unless they were actually run.

## Required validation

Run these gates after implementation:

```bash
uv run ruff check .
uv run ruff format --check .
uv run pyright
uv run pytest -q
uv run muzero --help
uv run muzero train --config configs/cartpole_muzero.yaml --profile cpu_debug --seed 0
uv run muzero checkpoint inspect runs/*/checkpoints/final.pt
uv run muzero benchmark --config configs/cartpole_muzero.yaml --component inference --steps 10
uv run muzero benchmark --config configs/cartpole_muzero.yaml --component search --steps 10
```

Also run targeted tests:

```bash
uv run pytest tests/integration/test_pendulum_sampled_smoke.py -q
uv run pytest tests/integration/test_gumbel_smoke.py -q
uv run pytest tests/integration/test_three_player_toy.py -q
uv run pytest tests/integration/test_checkpoint_roundtrip.py -q
```

## Stop conditions

Stop and report only if:

- validation fails after two self-repair attempts;
- a plan-level technical decision is missing;
- an external resource is required;
- `docs/plan.md` conflicts irreconcilably with the current code.

## Completion report

When done, report:

1. changed files;
2. module completion status;
3. exact validation command results;
4. unresolved issues, if any.
